libcudacxx/include/cuda/discard_memory

```diff
@@ -13,11 +13,17 @@
 
 #include <cuda/std/cstdint>
 #include <cuda/std/detail/__config>
+#include <cuda/std/detail/libcxx/include/__cuda/ptx.h>
 
 _LIBCUDACXX_BEGIN_NAMESPACE_CUDA
 
 inline _LIBCUDACXX_HOST_DEVICE void discard_memory(volatile void* __ptr, std::size_t __nbytes) noexcept
 {
+  // The discard PTX instruction is only available with PTX ISA 7.4 and later
+#if __cccl_ptx_isa < 740ULL
+  (void) (__ptr);
+  (void) (__nbytes);
+#else
```

Comment:
For larger conditionals we try to add comments to clearly label the different secions
```suggestion
#else // ^^^ No ISA 7.4 ^^^ / vvv With ISA 7.4 vvv
```